Mage/src/main/java/mage/watchers/common/PermanentsEnteredBattlefieldYourLastTurnWatcher.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package mage.watchers.common;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.UUID;
+import mage.constants.WatcherScope;
+import mage.game.Game;
+import mage.game.events.GameEvent;
+import mage.game.permanent.Permanent;
+import mage.watchers.Watcher;
+
+/**
+ *
+ * @author LevelX2 (spjspj)
+ */
+public class PermanentsEnteredBattlefieldYourLastTurnWatcher extends Watcher {
+
+    private final HashMap<UUID, List<Permanent>> enteringBattlefield = new HashMap<>();
+    private final HashMap<UUID, List<Permanent>> enteringBattlefieldLastTurn = new HashMap<>();
+    private final Map<UUID, Integer> activePlayer = new HashMap<>();
```

Comment:
Wouldn't here work a simple UUID of the last active player?

Reply:
Hi @LevelX2 I can't get the active player when it's reset as game is not passed into the reset function.